lib/capybara/queries/text_query.rb

```diff
@@ -31,6 +32,16 @@ def failure_message
           message << " but found #{@count} #{Capybara::Helpers.declension('time', 'times', @count)}"
         end
         message << " in #{@actual_text.inspect}"
+
+        if @node and visible?
```

Comment:
I started off with the total but decided the message was clearer by subtracting the visible hits from the total hits to make it clear that the invisible count was different. Got a better way of phrasing it?

I agree about not mentioning invisible hits in the negative case, I guess... It's hard to think through all the cases.